src/tools/scripts/run_and_test_website.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+######################################
+## Custom Web Almanac script        ##
+######################################
+#
+# This script installs all the required dependencies needed to run the
+# Web Almanac website providing you have python and node installed.
+#
+# It also runs our tests to ensure the website is working for all pages.
+#
+# It is used by various GitHub actions to build and test the site.
+#
+
+# exit when any command fails instead of trying to continue on
+set -e
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src
+fi
+
+echo "Installing and testing python environment"
+python -m pip install --upgrade pip
+pip install -r requirements.txt
+pytest
+
+echo "Installing node modules"
+npm install
+
+echo "Building website"
+npm run generate
+
+echo "Starting website"
+python main.py background &
+# Check website is running as won't have got feedback as backgrounded
+# use [p]ython so we don't match the grep itself
+ps -ef | grep "[p]ython main.py"
```

Comment:
This is just a quick check to avoid running the 100+ checks if the server dies immediately on start up due to an error. It's fine if the next step fails so this is not critical, but I'd rather see this one error, and 100+ errors. Had a typo and notice this so added it and think it's a good little check.

Reply:
I was wondering, why do we have the "p" of python in brackets here?

Reply:
I think I got it. It is to avoid the spurious line of the grep itself.

Reply:
Yup - As it says in the line above 😀
```sh
# use [p]ython so we don't match the grep itself
```

Reply:
If only I read the inline comment! :laughing: